internal/sdkprovider/service/vpc/gcp_org_vpc_peering_connection.go

```diff
@@ -0,0 +1,196 @@
+		ForceNew:    true,
+		Required:    true,
+		Type:        schema.TypeString,
+		Description: userconfig.Desc("Google Cloud VPC network name.").ForceNew().Build(),
+	},
+	"state": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "State of the peering connection.",
+	},
+	"self_link": {
+		Computed:    true,
+		Type:        schema.TypeString,
+		Description: "Computed Google Cloud network peering link.",
+	},
+}
+
+func ResourceGCPOrgVPCPeeringConnection() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Creates and manages a Google Cloud VPC peering connection.",
+		CreateContext: common.WithGenClientDiag(resourceGCPOrgVPCPeeringConnectionCreate),
+		ReadContext:   common.WithGenClientDiag(resourceGCPOrgVPCPeeringConnectionRead),
+		DeleteContext: common.WithGenClientDiag(resourceGCPOrgVPCPeeringConnectionDelete),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: schemautil.DefaultResourceTimeouts(),
+
+		Schema: aivenGCPOrgVPCPeeringConnectionSchema,
+	}
+}
+
+func resourceGCPOrgVPCPeeringConnectionCreate(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	var (
+		orgID        = d.Get("organization_id").(string)
+		vpcID        = d.Get("organization_vpc_id").(string)
+		gcpProjectID = d.Get("gcp_project_id").(string)
+		peerVPC      = d.Get("peer_vpc").(string)
+
+		req = organizationvpc.OrganizationVpcPeeringConnectionCreateIn{
+			PeerCloudAccount: gcpProjectID,
+			PeerVpc:          peerVPC,
+		}
+	)
+
+	pCon, err := createPeeringConnection(ctx, orgID, vpcID, client, d, req)
+	if err != nil {
+		return diag.Errorf("Error creating VPC peering connection: %s", err)
+	}
+
+	diags := getDiagnosticsFromState(newOrganizationVPCPeeringState(pCon))
+
+	d.SetId(schemautil.BuildResourceID(orgID, vpcID, pCon.PeerCloudAccount, pCon.PeerVpc))
+
+	// in case of an error delete VPC peering connection
+	if diags.HasError() {
+		deleteDiags := resourceAzureOrgVPCPeeringConnectionDelete(ctx, d, client)
+		d.SetId("") // Clear the ID after delete
+
+		return append(diags, deleteDiags...)
+	}
+
+	return append(diags, resourceGCPOrgVPCPeeringConnectionRead(ctx, d, client)...)
+}
+
+func resourceGCPOrgVPCPeeringConnectionRead(ctx context.Context, d *schema.ResourceData, client avngen.Client) diag.Diagnostics {
+	orgID, vpcID, cloudAcc, peerVPC, err := schemautil.SplitResourceID4(d.Id())
+	if err != nil {
+		return diag.Errorf("error parsing GCP peering VPC ID: %s", err)
+	}
+
+	vpc, err := client.OrganizationVpcGet(ctx, orgID, vpcID)
+	if err != nil {
+		if avngen.IsNotFound(err) {
+			return diag.FromErr(schemautil.ResourceReadHandleNotFound(err, d))
+		}
+
+		return diag.Errorf("failed to get VPC with ID %q: %s", vpcID, err)
+	}
+
+	pCon := lookupGCPPeeringConnection(vpc, cloudAcc, peerVPC)
+	if pCon == nil {
+		return diag.FromErr(schemautil.ResourceReadHandleNotFound(errors.New("VPC peering connection not found"), d))
```

Comment:
`ResourceReadHandleNotFound` wont't recognize the error

Reply:
Fixed